### PR TITLE
Add cancel / create button on fileMenu form

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Description
Enhancement: Show create and cancel buttons in the 'New file menu'

If the user hits the '+' button in the UI, a context menu will be shown,
where the user is able to choose for example 'Directory'.
A form will show up to set the new directory name, the user needs to confirm
by hitting the enter key.
This might not been understood by every user at the first glance.
Therefore, with this PR a create and cancel button has been added.

## Related Issue

- Fixes https://github.com/owncloud/enterprise/issues/4684

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by JanAckermann** [comment]
TODO:
- As enter cofirms form, escape should close the form -

**Comment by tbsbdr** [comment]
some visual guidance would be nice, i.e. visualize "create" as the primary action and cancel as the secondary action. I could not find a secondary-styled button in oc classic, but in the "create link" dialog we have a blue primary button which would have the same effect.

**--> could you use the blue styling for the create-button?**

image

**Comment by JanAckermann** [comment]
@tbsbdr i already updated this pr with blue confirm button

**Comment by tbsbdr** [comment]
Works like a charm, thanks for the fix!

**Comment by JanAckermann** [comment]
@mmattel at first glance, we don't need a docs issue

**Comment by JanAckermann** [comment]
@phil-davis should we create acceptance tests or are the js tests already satisfying ?

**Comment by phil-davis** [comment]
@JanAckermann the current acceptance tests "hit enter" to create a folder (or a text file).
We should add a scenario that clicks the "Create"  button. And a scenario that clicks the "Cancel" button.

Do you want to do that? Or will I get someone to do it?

(Also, I see that the "Text File" option also has the new Cancel and Create buttons. Similar scenarios could be added to files_texteditor to test those)

**Comment by JanAckermann** [comment]
> Do you want to do that? Or will I get someone to do it?
We can do it.

> (Also, I see that the "Text File" option also has the new Cancel and Create buttons. Similar scenarios could be added to files_texteditor to test those)

From technically POV this is core functionality and behaves the same on text file and so on, IHMO no seperated tests for apps needed

**Comment by phil-davis** [comment]
Fails `make test-php-style`
https://drone.owncloud.com/owncloud/core/31563/2/7

**Comment by JammingBen** [comment]
Question is why, locally everything works fine 🤔

**Comment by phil-davis** [comment]
The extra style checker for the acceptance test code does not have an automatic fixer.

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiCapabilities-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31575/55/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiProvisioningGroups-v2-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31575/62/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiAuthOcs-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31575/53/1

**Comment by JammingBen** [comment]
Okay we really need to fix https://github.com/owncloud/core/issues/39046, this starts to get annoying...

**Comment by phil-davis** [comment]
> Okay we really need to fix #39046, this starts to get annoying...

I restarted CI again.

Who can fix #39046 ?

**Comment by JammingBen** [comment]
> Who can fix #39046 ?

Doing some tests via https://github.com/owncloud/core/pull/39069 to determine the issue

File: changelog/unreleased/39056 [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Enhancement: Show create and cancel buttons in the 'New file menu'
+
+If the user hits the '+' button in the UI, a context menu will be shown,
+where the user is able to choose for example 'Directory'.
+A form will show up to set the new directory name, the user needs to confirm
```
Comment (phil-davis) [inline changelog/unreleased/39056:5]:
```suggestion
where the user is able to choose for example 'Folder'.
A form will show up to set the new folder name, the user needs to confirm
```

File: tests/acceptance/features/lib/FilesPageElement/NewFileMenu.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+use ast\Node;
+use Behat\Mink\Session;
+use Behat\Mink\Element\NodeElement;
+use Page\OwncloudPage;
+use SensioLabs\Behat\PageObjectExtension\PageObject\Exception\ElementNotFoundException;
+
+/**
+ * Object for files action Menu on the FilesPage
+ * containing actions like Rename, Delete, etc
```
Comment (phil-davis) [inline tests/acceptance/features/lib/FilesPageElement/NewFileMenu.php:35]: This looks like copy-paste from another page object

**Review by phil-davis — COMMENTED** [review]
Code works

File: changelog/unreleased/39056 (hunk 1 above)
Comment (phil-davis) [reply changelog/unreleased/39056:5]: IMO better to use "Folder" here, because that is the word used on the UI

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
Looks good - you could squash the 8 commits.

**Comment by JanAckermann** [comment]
Thanks, squashed into two, for easy patching

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliBackground-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/119/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiWebdavLocks2-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/94/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiAuthWebDav-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/54/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliProvisioning-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/122/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiProvisioning-v2-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/60/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiMain-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/58/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiTranslation-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/87/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiSharees-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/67/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiWebdavUpload1-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/103/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiShareManagementBasicToShares-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/71/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiWebdavProperties2-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/102/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiWebdavPreviews-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/100/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiSharePublicLink1-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/76/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiShareOperationsToShares2-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/75/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiShareReshareToShares3-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/83/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiFederationToShares1-10.6.0-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/115/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiFederationToRoot1-git-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/107/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiFederationToShares1-latest-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/114/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliMain-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/121/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliAppManagement-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31597/124/1

**Review by tbsbdr — APPROVED** [review]
lgtm - thanks!!

**Pull request merged**